vm/devices/storage/disk_nvme/nvme_driver/src/queue_pair.rs

```diff
@@ -216,7 +216,10 @@ impl<T: AerHandler> QueuePair<T> {
             } else {
                 PER_QUEUE_PAGES_NO_BOUNCE_BUFFER * PAGE_SIZE
             };
-        let dma_client = device.dma_client();
+
+        let dma_client = device
+            .persistent_dma_client()
+            .unwrap_or(device.ephemeral_dma_client());
```

Comment:
Pragmatically this behaves the same as what is checked in already. But, for NVMe, I think we want to control which DMA allocator is used at the nvme_manager layer. That is the component controlling policy. Ideally ephemeral_dma_client is `None` when we have a private pool, for example.